mico-core/src/main/java/io/github/ust/mico/core/broker/MicoApplicationBroker.java

```diff
@@ -348,6 +343,74 @@ public MicoServiceDeploymentInfo addKafkaFaasConnectorInstanceToMicoApplicationB
+        throws MicoApplicationNotFoundException, MicoApplicationIsNotUndeployedException, KafkaFaasConnectorVersionNotFoundException,
+        KafkaFaasConnectorInstanceNotFoundException, KafkaFaasConnectorInstanceAlreadyIncludedWithSameVersionInMicoApplicationException {
+
+        // Retrieve application and service from database (checks whether they exist)
+        MicoApplication micoApplication = getMicoApplicationByShortNameAndVersion(applicationShortName, applicationVersion);
+
+        // Check whether application is currently undeployed, if not it is not allowed to add a KafkaFaasConnector instance to the application
+        checkIfApplicationIsUndeployed(micoApplication);
+
+        MicoService kfConnectorWithRequestedVersion = getKafkaFaasConnectorServiceByVersion(kfConnectorVersion);
+
+        // Find all service deployment information with identical instance ID within this application
+        List<MicoServiceDeploymentInfo> kfConnectorsWithSameInstanceId = micoApplication.getKafkaFaasConnectorDeploymentInfos().stream()
+            .filter(kfCDI -> kfCDI.getInstanceId().equals(instanceId)).collect(Collectors.toList());
+        log.debug("Found {} KafkaFaasConnector(s) with instance ID '{}' within application '{}' '{}'.",
+            kfConnectorsWithSameInstanceId.size(), instanceId, applicationShortName, applicationVersion);
+
+        if (kfConnectorsWithSameInstanceId.isEmpty()) {
+            throw new KafkaFaasConnectorInstanceNotFoundException(instanceId);
+        }
+        if (kfConnectorsWithSameInstanceId.size() > 1) {
+            // Illegal state, an instanceIdOptional must be unique
+            String errorMessage = "There are " + kfConnectorsWithSameInstanceId.size() +
+                " service deployment information stored for KafkaFaasConnector in version with the same instance ID '"
+                + instanceId + "' used by the application '" + micoApplication.getShortName() +
+                "' '" + micoApplication.getVersion() + "'. However, an instance ID must be unique.";
+            log.error(errorMessage);
+            throw new IllegalStateException(errorMessage);
+        }
+        // There is already one instance with the same id, replace its newer version...
+        MicoServiceDeploymentInfo existingKfConnectorSDI = kfConnectorsWithSameInstanceId.get(0);
+        log.debug("KafkaFaasConnector with instance ID '{}' and version '{}' is already used by the application '{}' '{}'. " +
+                "Replace it with the version '{}'.", existingKfConnectorSDI.getInstanceId(), existingKfConnectorSDI.getService().getVersion(),
+            applicationShortName, applicationVersion, kfConnectorVersion);
+
+        // ... but only replace if the new version is different from the current version
+        if (existingKfConnectorSDI.getService().getVersion().equals(kfConnectorVersion)) {
+            throw new KafkaFaasConnectorInstanceAlreadyIncludedWithSameVersionInMicoApplicationException(
```

Comment:
This makes it so that if the ui triggers the request two times by accident the user sees an error for the second request. As the request itself is idempotent why not implement it in an idempotent way?